### PR TITLE
Reserved numbers

### PR DESCRIPTION
**Pull request by VladyslavBanul**
https://app.asana.com/0/1201128813141482/1201275099482089/f

### CODE REVIEW

File: src/components/Table/components/TableActions.tsx [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
-import { TableActionsType } from "utils/types/tableConfig";
+import {
+  CustomActionType,
+  TableActionsType,
+  TableData,
+} from "utils/types/tableConfig";
 
 import { useStyles } from "./styles";
+import clsx from "clsx";
+import { Row } from "react-table";
```
Comment (SergeyKoliadintsev) [inline src/components/Table/components/TableActions.tsx:12]: Could you please add this import to lib imports

File: src/levels/subscription/components/SubscriptionNumbersPage/TabsContent/ReservedNumbers/styles.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { makeStyles } from "@material-ui/core";
+
+import { ThemeDefaultOptions } from "utils/types/themeConfig";
+//@ts-ignore
```
Comment (SergeyKoliadintsev) [inline src/levels/subscription/components/SubscriptionNumbersPage/TabsContent/ReservedNumbers/styles.ts:4]: you need this file ?

File: src/storage/singletons/Entitlements/index.ts [hunk 3]
```diff
@@ -42,6 +45,26 @@ class SubscriptionEntitlementsStore {
...
+          [curr.countryCode]: {
+            ...availableEntitlements[curr.countryCode],
+            [curr.numberType]: curr.entitlement - curr.assigned,
+          },
+        };
+      },
+      {},
+    );
+  }
+  setAvailable = (av: AvailableEntitlements) => {
```
Comment (SergeyKoliadintsev) [inline src/storage/singletons/Entitlements/index.ts:65]: rename this variable please

File: src/storage/singletons/Numbers/index.tsx [hunk 4]
```diff
@@ -113,6 +133,30 @@ class NumbersStore {
...
+      loaderName: "@postNumber",
+      method: "post",
+      payload,
+    })
+      .then(() => {
+        runInAction(() => {
+          this.getNumbersData(tenantID, subscriptionID);
+        });
+        successCallback && successCallback();
+        this.getReservedNumbers(tenantID, subscriptionID);
```
Comment (SergeyKoliadintsev) [inline src/storage/singletons/Numbers/index.tsx:153]: Could you please refactor that

File: src/levels/subscription/components/SubscriptionNumbersPage/TabsContent/ReservedNumbers/index.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+          numbers: [+row.original.nsn],
+        });
+      },
+    },
+    {
+      actionName: "info",
+      iconComponent: (
+        <Tooltip
+          arrow
+          title="No more entitlements for this number left"
```
Comment (viktorsmakhtin) [inline src/levels/subscription/components/SubscriptionNumbersPage/TabsContent/ReservedNumbers/index.tsx:97]: pls add locale

**Pull request merged**